src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java

```diff
@@ -449,37 +429,26 @@ public int compare(FeedbackSessionAttributes session1, FeedbackSessionAttributes
      * method is called with combined feedback sessions from many courses
      */
     public static void sortFeedbackSessionsByCreationTimeDescending(List<FeedbackSessionAttributes> sessions) {
-        Collections.sort(sessions, new Comparator<FeedbackSessionAttributes>() {
-            @Override
-            public int compare(FeedbackSessionAttributes session1, FeedbackSessionAttributes session2) {
-                int result = session2.createdTime.compareTo(session1.createdTime);
-                if (result == 0) {
-                    if (session1.endTime == null || session2.endTime == null) {
-                        if (session1.endTime == null) {
-                            --result;
-                        }
-                        if (session2.endTime == null) {
-                            ++result;
-                        }
-                    } else {
-                        result = session2.endTime.compareTo(session1.endTime);
-                    }
-                }
-
-                if (result == 0) {
-                    result = session2.startTime.compareTo(session1.startTime);
+        sessions.sort(Comparator.comparing((FeedbackSessionAttributes session) -> session.startTime).reversed()
+                .thenComparing(session -> session.courseId)
+                .thenComparing(session -> session.feedbackSessionName));
+        sessions.sort((session1, session2) -> {
```

Comment:
Now that the end time comparator is isolated, this can be simplified to a large extent. In particular, try your best so that you do not even need to use the variable `result`.